src/main/java/StringManipulation.java

```diff
@@ -1,33 +1,78 @@
-public class StringManipulation implements StringManipulationInterface {
+public class StringManipulation {
+
+    private String string;
 
-    @Override
     public String getString() {
-        return null;
+        return string;
     }
 
-    @Override
     public void setString(String string) {
+        this.string = string;
     }
 
-    @Override
     public int count() {
-        return 0;
+        if (string == null || string.isEmpty()) {
+            return 0;
+        }
+        String[] words = string.trim().split("\\s+");
+        return words.length;
     }
 
-    @Override
     public String removeNthCharacter(int n, boolean maintainSpacing) {
-        return null;
+        if (n <= 0) {
+            throw new IllegalArgumentException("Invalid value for n");
+        }
+        if (string == null || string.isEmpty() || n > string.length()) {
+            throw new IndexOutOfBoundsException("n is greater than the string length");
+        }
+
+        StringBuilder result = new StringBuilder();
+        for (int i = 0; i < string.length(); i++) {
+            if ((i + 1) % n != 0) {
+                result.append(string.charAt(i));
+            } else if (maintainSpacing) {
+                result.append(' ');
+            }
+        }
+        return result.toString();
     }
 
-    @Override
     public String[] getSubStrings(int startWord, int endWord) {
-        return null;
-    }
+        if (startWord <= 0 || endWord <= 0 || startWord > endWord) {
+            throw new IllegalArgumentException("Invalid startWord or endWord values");
+        }
+        if (string == null || string.isEmpty()) {
+            throw new IndexOutOfBoundsException("String is empty");
+        }
 
-    @Override
-    public String restoreString(int[] indices) {
-        return null;
+        String[] words = string.trim().split("\\s+");
+        if (endWord > words.length) {
+            throw new IndexOutOfBoundsException("String has fewer words than endWord");
+        }
+
+        String[] substrings = new String[endWord - startWord + 1];
+        int index = 0;
+        for (int i = startWord - 1; i < endWord; i++) {
+            substrings[index] = words[i];
+            index++;
+        }
+        return substrings;
     }
 
+    public String restoreString(int[] indices) {
+        if (string == null || indices == null || string.length() != indices.length) {
```

Comment:
Java already throws a runtime error if you try to call `restoreString()` with an uninitialized int array, so I don't think you need the second condition here.